### PR TITLE
PP-7488 Add unit test to ensure authorisation middleware used

### PR DESCRIPTION
**Pull request by stephencdaly**
Add test to ensure that authorisation middleware is called for all routes that require authentication.

Achieve this by specifying in the test which paths do not require authentication. Then check all route setup calls to ensure that the authorisation middleware is used for routes with paths that are not in this list.

This will ensure that the middleware is used for any new routes that are registered, unless they are explicitly added to the exclude list.

When the new authentication middleware has been added, will modify to check if the new middleware is run instead. And can be updated to remove checks for the old middleware when we have fully replaced it.

### CODE REVIEW

File: test/unit/routes.test.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    // We currently call `app.use` to specify a list of paths to use certain middleware for, 
+    // including the authorisation middleware. We need to check that paths are either included in
+    // the array for this call, or the middleware is added to the stack individually for the route
+    const authenticatedPathsArg = app.use.getCalls()
+      .find(call => {
+        return Array.isArray(call.args[0]) &&
+          call.args.includes(oldAuthorisationMiddleware)
+      })
+      .args[0]
+    const authenticatedPaths = flattenPaths(authenticatedPathsArg)
```
Comment (sfount) [inline test/unit/routes.test.js:64]: It took me a minute to fully get what was going on with this step before getting to the routes! I think this edge case coverage could also be removed if we're encouraging route based authorisation with the new auth middleware

**Review by sfount — APPROVED** [review]
Clear, concise and covers a weak point in app authorisation that has already caught paths that were misconfigured.

This is an interesting and well executed idea which I think would improve the confidence in security in all of the public facing apps 👍

File: test/unit/routes.test.js (hunk 1 above)
Comment (stephencdaly) [reply test/unit/routes.test.js:64]: Yep, it can be removed down the line, maybe we should put a reminder in to do it on another story?
